projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,316 @@
+
+    /**
+     * Reads Html template's content to String. (Template's path stored at 'templatePath' class variable)
+     * @return The String holding lawsuit's template's content in an Html format.
+     */
+    protected String readTemplate() {
+
+        StringBuilder contentBuilder = new StringBuilder();
+        try {
+            BufferedReader in = new BufferedReader(new FileReader(lawsuitTemplateFilePath));
+            String str;
+            while ((str = in.readLine()) != null) {
+                contentBuilder.append(str);
+            }
+            in.close();
+        } catch (IOException e) {
+        }
+        String content = contentBuilder.toString();
+        return content;
+    }
+
+    /**
+     * Fill lawsuit template with user & spam company's details.
+     * @return Template's content as Html, with fields replaced.
+     */
+    protected String fillInTemplate() {
+
+        // Read template's content:
+        String res = templateContent;
+
+        // Fill in general form details:
+        claimCase = sentHaser ? claimCaseHaser: claimCaseSubscription;
+        fiveLawsuitsStatus = sentMoreThanFiveLawsuits ? moreThanFiveLawsuits: lessThanFiveLawsuits;
+        res = res.replace("&lt;claimCase&gt;", claimCase);
+        res = res.replace("&lt;claimAmount&gt;", claimAmount);
+        res = res.replace("&lt;spamType&gt;", spamType);
+        res = res.replace("&lt;moreThanFiveLawsuits&gt;", fiveLawsuitsStatus);
+        res = res.replace("&lt;undefined&gt;", "");
+        res = res.replace("&lt;receivedSpamDate&gt;", dateFormatter.format(receivedSpamDate));
+        res = res.replace("&lt;currentDate&gt;", dateFormatter.format(currentDate));
+
+        // User details:
+        res = res.replace("&lt;userFullName&gt;", userPrivateName + " " + userLastname);
+        res = res.replace("&lt;userId&gt;", userID);
+        res = res.replace("&lt;userAddress&gt;", userAddress);
+        res = res.replace("&lt;userPhone&gt;", userPhone);
+        res = res.replace("&lt;userFax&gt;", userFax);
+
+        // First company's details:
+        res = res.replace("&lt;companyName&gt;", companyName);
+        res = res.replace("&lt;companyId&gt;", companyId);
+        res = res.replace("&lt;companyAddress&gt;", companyAddress);
+        res = res.replace("&lt;companyPhone&gt;", companyPhone);
+        res = res.replace("&lt;companyFax&gt;", companyFax);
+
+        // Second company's details:
+        res = res.replace("&lt;company2Name&gt;", company2Name);
+        res = res.replace("&lt;company2Id&gt;", company2Id);
+        res = res.replace("&lt;company2Address&gt;", company2Address);
+        res = res.replace("&lt;company2Phone&gt;", company2Phone);
+        res = res.replace("&lt;company2Fax&gt;", company2Fax);
+
+        return res;
+    }
+
+    /**
+     * Responsible for the output lawsuit PDF file name format.
+     * @return File name with ".pdf" extension.
+     */
+    protected String createPdfLawsuitFileName(){
```

Comment:
When methods are short and don't do much, we usually call them `get`. `create` and `build` is for heavier stuff.
So I suggest renaming to `getPdfLawsuitFileName`.

Reply:
Done.